lib/ZnapZend/Config.pm

```diff
@@ -80,15 +80,30 @@ my $checkBackupSets = sub {
 
     for my $backupSet (@{$self->backupSets}){
 
-        # in case there is only one property on this dataset, which is the "enabled" and is set to "off"
-        # consider it a normal situation and do not even notify it. This situation will appear
-        # when there are descendants of recursive ZFS dataset that should be skipped.
-        # Note: backupSets will have at least the key "Src". Therefore, we need to skip the
-        # dataset if there are two properties and one of them is "enabled".
+        # In case there is only one property on this dataset, which is the
+        # "enabled" flag and is set to "off"; consider it a normal situation
+        # and do not even notify it. This situation will appear when there
+        # are descendants of recursive ZFS dataset that should be skipped.
+        # Note: backupSets will have at least the key "src". Therefore, we
+        # need to skip the dataset if there are two properties and one of
+        # them is "enabled".
         if (keys(%{$backupSet}) eq 2 && exists($backupSet->{"enabled"})){
            next;
         }
 
+        if ( $backupSet->{src} =~ m/[\@]/ ) {
+            # If we are here, somebody fed us a snapshot in the list of
+            # datasets, which is likely a bug elsewhere in discovery.
+            # We do not want to fail whole backup below due to faulted
+            # dataSetExists() below, so just ignore this entry.
+            # If we really do get here, take a hard look at recursive
+            # and/or inherited modes for run-once.
+            print STDERR "#checkBackupSets# SKIP backupSet='" . $backupSet->{src} .
```

Comment:
not using zLog here ?

Reply:
I thought about it a little... this PR is essentially a `git revert` + typo fix, so I did not go too deep. Given that this catches a situation that should never happen at all, I am not sure what could be proper logging here.

Reply:
well the kind of logging that is best suited to catch the attendents attention ... zLog->error I would suggest ... after all they should be watching the znapzend logs

Reply:
Done, checked that `$self->zLog->error()` by itself is not fatal to code progressing ;)